app/helpers/authentication_helper.rb

```diff
@@ -15,8 +15,11 @@ def authentication_enabled_providers
     end
   end
 
-  def current_user_authentication_enabled_providers
-    providers = Authentication::Providers.enabled & current_user.identities.pluck(:provider).map(&:to_sym)
+  # Returns the list of enabled providers for the given user, defaults to `current_user`
+  def authentication_enabled_providers_for_user(user = nil)
```

Comment:
Ruby's default arguments are quite flexible, you can call other methods there. This way you don't need manually default to `current_user` on the next line.

```suggestion
  def authentication_enabled_providers_for_user(user = current_user)
```